eclair-core/src/main/scala/fr/acinq/eclair/json/JsonSerializers.scala

```diff
@@ -118,6 +118,34 @@ object ByteVector32KmpSerializer extends MinimalSerializer({
   case x: fr.acinq.bitcoin.ByteVector32 => JString(x.toHex)
 })
 
+object TxIdSerializer extends MinimalSerializer({
+  case x: TxId => JString(x.value.toHex)
+})
+
+object TxIdKeySerializer extends MinimalKeySerializer({
+  case x: TxId => x.value.toHex
```

Comment:
This is a key serializer, for usage in a `Map` so this isn't a `JString`, this must be a string!

> I would have defined a TxId.toHex and TxHash.toHex.

Actually the `toString` method on `TxId` or `TxHash` already does that.

Reply:
> This is a key serializer, for usage in a Map so this isn't a JString, this must be a string!

I stand corrected!

> should I change this to use toString instead of .value.toHex?

No, `toString` is too generic, and doesn't work in the general case for `ByteVector`s. I think it must be `toHex`, which has strong semantics. This is very minor.